SQLFrontend/CommandFunctors.h

```diff
@@ -0,0 +1,387 @@
+          key_with_spec += " REFERENCES ";
+          const std::string foreign_table = key_with_spec_json["foreign_table"].GetString();
+          const std::string foreign_column = key_with_spec_json["foreign_column"].GetString();
+          key_with_spec += foreign_table + "(" + foreign_column + ")";
+        } else {
+          CHECK(type == "SHARD KEY");
+        }
+        keys_with_spec.push_back(key_with_spec);
+      }
+      return keys_with_spec;
+    }
+}
+
+StandardCommand( ListColumns, {
+    decltype( p[1] )& table_name( p[1] );
+
+    auto on_success_lambda = [&]( ContextType& context ) {
+      const auto table_details = context.table_details;
+      if (table_details.view_sql.empty()) {
+        std::string temp_holder(" ");
+        if (table_details.is_temporary) {
+          temp_holder = " TEMPORARY ";
+        }
+        output_stream << "CREATE" + temp_holder + "TABLE " + table_name + " (\n";
+      } else {
+        output_stream << "CREATE VIEW " + table_name + " AS " + table_details.view_sql << "\n";
+        output_stream << "\n"
+                  << "View columns:"
+                  << "\n\n";
+      }
+      std::string comma_or_blank("");
+      for (TColumnType p : table_details.row_desc) {
+        if (p.is_system) {
+          continue;
+        }
+        std::string encoding;
+        if (p.col_type.type == TDatumType::STR) {
+          encoding =
+              (p.col_type.encoding == 0 ? " ENCODING NONE" : " ENCODING " + thrift_to_encoding_name(p.col_type) + "(" +
+                                                                 std::to_string(p.col_type.comp_param) + ")");
+  
+        } else {
+          encoding = (p.col_type.encoding == 0 ? "" : " ENCODING " + thrift_to_encoding_name(p.col_type) + "(" +
+                                                          std::to_string(p.col_type.comp_param) + ")");
+        }
+        output_stream << comma_or_blank << p.col_name << " " << thrift_to_name(p.col_type)
+                  << (p.col_type.nullable ? "" : " NOT NULL") << encoding;
+        comma_or_blank = ",\n";
+      }
+      if (table_details.view_sql.empty()) {
+        const auto keys_with_spec = unserialize_key_metainfo(table_details.key_metainfo);
+        for (const auto& key_with_spec : keys_with_spec) {
+          output_stream << ",\n" << key_with_spec;
+        }
+        // push final ")\n";
+        output_stream << ")\n";
+        comma_or_blank = "";
+        std::string frag = "";
+        std::string page = "";
+        std::string row = "";
+        if (DEFAULT_FRAGMENT_ROWS != table_details.fragment_size) {
+          frag = "FRAGMENT_SIZE = " + std::to_string(table_details.fragment_size);
+          comma_or_blank = ", ";
+        }
+        if (table_details.shard_count) {
+          frag += comma_or_blank + "SHARD_COUNT = " + std::to_string(table_details.shard_count);
+          comma_or_blank = ", ";
+        }
+        if (DEFAULT_PAGE_SIZE != table_details.page_size) {
```

Comment:
ibid

Reply:
ibid?  (Going through the remarks fixing things.)